src/controlplane/client/connector.go

```diff
@@ -0,0 +1,274 @@
+		break
+	case strings.EqualFold(auth.Type, mTLSAuth):
+		return validateMTLS(auth.MTLS)
+	default:
+		return fmt.Errorf("authorization type not supported: %q", auth.Type)
+	}
+
+	return nil
+}
+
+func validateMTLS(mTLS *config.MTLS) error {
+	if mTLS == nil {
+		return fmt.Errorf("mTLS config must exist")
+	}
+
+	if mTLS.TLSSecretName == "" {
+		return fmt.Errorf("TLSSecretName cannot be empty")
+	}
+	return nil
+}
+
+func (dp *defaultConnector) newHTTPClient(endpoint config.Endpoint) (*http.Client, error) {
+	client := &http.Client{Timeout: DefaultTimout}
+
+	// Here we're using the default http.Transport configuration, but with a modified TLS config.
+	// For some reason the DefaultTransport is casted to an http.RoundTripper interface, so we need to convert it back.
+	t := http.DefaultTransport.(*http.Transport).Clone()
+	t.TLSClientConfig = &tls.Config{InsecureSkipVerify: endpoint.InsecureSkipVerify}
+	client.Transport = t
+
+	if err := dp.configureAuthentication(client, endpoint); err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (dp *defaultConnector) configureAuthentication(httpClient *http.Client, endpoint config.Endpoint) error {
+	if endpoint.Auth == nil {
+		return nil
+	}
+
+	switch {
+	case endpoint.Auth.Type == plainAuth:
+		dp.logger.Debugf("Plain auth is used for endpoint %q", endpoint.URL)
+
+	case strings.EqualFold(endpoint.Auth.Type, bearerAuth):
+		dp.logger.Debugf("kubernetes Bearer token auth is used for endpoint %q", endpoint.URL)
+		httpClient.Transport = transport.NewBearerAuthRoundTripper(dp.inClusterConfig.BearerToken, httpClient.Transport)
+
+	case strings.EqualFold(endpoint.Auth.Type, mTLSAuth):
+		tlsConfig, err := dp.getTLSConfigFromSecret(endpoint.Auth.MTLS)
+		if err != nil {
+			return fmt.Errorf("could not load TLS configuration: %w", err)
+		}
+
+		dp.logger.Debugf("MTLS auth is used for endpoint %q", endpoint.URL)
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+
+	default:
+		return fmt.Errorf("authorization type not supported: %q", endpoint.Auth.Type)
+	}
+
+	return nil
+}
+
+func (dp *defaultConnector) getTLSConfigFromSecret(mTLSConfig *config.MTLS) (*tls.Config, error) {
+	if mTLSConfig == nil {
+		return nil, fmt.Errorf("mTLS config cannot be nil")
+	}
+
+	if mTLSConfig.TLSSecretName == "" {
+		return nil, fmt.Errorf("mTLS secret name cannot be empty")
+	}
+
+	namespace := mTLSConfig.TLSSecretNamespace
+	secretName := mTLSConfig.TLSSecretName
+	if namespace == "" {
+		dp.logger.Debugf("TLS Secret name configured, but not TLS Secret namespace. Defaulting to `default` namespace.")
+		namespace = "default"
+	}
+
+	secret, err := dp.kc.CoreV1().Secrets(namespace).Get(context.Background(), secretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not find secret %q containing TLS configuration: %w", secretName, err)
+	}
+
+	var cert, key, cacert []byte
+
+	var ok bool
+	if cert, ok = secret.Data["cert"]; !ok {
```

Comment:
We are unifying the use cases of mTLS and TLS, I believe that right now it is not possible to configure an endpoint with normal HTTPS. Do we want that?